src/sentry/static/sentry/app/views/organizationIntegrations/integrationIcon.tsx

```diff
@@ -12,16 +12,40 @@ type Props = {
 
 type IconProps = Pick<Props, 'size'>;
 
-const Icon = styled('img')<IconProps>`
+const StyledIcon = styled('img')<IconProps>`
   height: ${p => p.size}px;
   width: ${p => p.size}px;
   border-radius: 2px;
   display: block;
 `;
 
+class Icon extends React.Component<Props> {
+  state = {
+    imgSrc: '',
+  };
+
+  componentDidMount() {
+    const {integration} = this.props;
+    this.setState({imgSrc: integration.icon});
+  }
+  render() {
+    const {integration, size} = this.props;
+
+    return (
+      <StyledIcon
+        size={size}
+        src={this.state.imgSrc}
+        onError={_ => {
```

Comment:
@NisanthanNanthakumar Nit: if you aren't going to use the event in the callback, you can always use a callback with 0 arguments.

Reply:
Sorry my golang is mixing in with my javascript